flake8_async/runner.py

```diff
@@ -25,18 +26,17 @@
 
     from libcst import Module
 
-    from .base import Error, Options
     from .visitors.flake8asyncvisitor import Flake8AsyncVisitor, Flake8AsyncVisitor_cst
 
 
 @dataclass
 class SharedState:
     options: Options
-    problems: list[Error] = field(default_factory=list)
-    noqas: dict[int, set[str]] = field(default_factory=dict)
+    problems: list[Error] = field(default_factory=list[Error])
+    noqas: dict[int, set[str]] = field(default_factory=dict[int, set[str]])
```

Comment:
huh, it seems like pyright needing this duplication of types might be an upstream issue?

Reply:
![2025-04-17T22:30:41,090407337+02:00](https://github.com/user-attachments/assets/1e42cd30-de51-4d2f-91d8-fc86400d7425)
intended... https://github.com/microsoft/pyright/issues/10277#issuecomment-2786881091

so yeah hence me opening https://github.com/microsoft/pyright/discussions/10336